react-components/src/data/details.js

```diff
@@ -0,0 +1,11 @@
+import BasicDetails from "./details/BasicDetails";
+import InfoTypeSelectionList from "./details/InfoTypeSelectionList"
+import Tag from "./details/Tag"
+import I18N from "../util/I18NWrapper";
+import intlData from "../data/messages";
+
+const BasicDetailsWithI18N = I18N(BasicDetails, intlData);
```

Comment:
This is probably just a preference, but I named all of my components `MyComponentBase`.  Then in this file I imagined we might eventually end up wrapping our components in several different HOCs, and at the end export `MyComponent`.  I just thought it made the names cleaner, but it doesn't really matter.

Reply:
I differ here:

1. Naming component as `MyComponentBase` gives an impression that the `Component` is base component and something else extends this. But what we are doing is composition and not extend.

2. In my case not I didn't need to wrap all the components with `I18N` HOC. I wrap the parent component and pass `I18N` as props to its children.

3. I think I like what the MUI does in its examples. Instead of wrapping the component and exporting it here with a Different name, we can wrap it with I18N when actually export the component
I my case this will be export default withI18N(BasicDetails, intlData); //Inside BasicDetails.js

Reply:
Sure, that works too 👍